In [1]:
# load library
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

from IPython.display import display
pd.options.display.max_rows = None
pd.options.display.max_columns = None

In [2]:
# import data
calendar = pd.read_csv('calendar.csv')
listings = pd.read_csv('listings.csv')
reviews = pd.read_csv('reviews.csv')

# Explore what info calendar contain

In [4]:
calendar.shape

(1308890, 4)

In [28]:
calendar.head()

listing_id        date available price
0    12147973  2017-09-05         f   NaN
1    12147973  2017-09-04         f   NaN
2    12147973  2017-09-03         f   NaN
3    12147973  2017-09-02         f   NaN
4    12147973  2017-09-01         f   NaN

In [12]:
calendar.listing_id.nunique()

3585

In [8]:
calendar.listing_id.value_counts()

12898806    730
14710769    365
7018497     365
3946215     365
951473      365
5718921     365
3278624     365
13624476    365
9626428     365
12175961    365
11370704    365
14585486    365
14591627    365
14235257    365
10053235    365
13233250    365
6291807     365
4369675     365
5729203     365
14168664    365
6005290     365
2862120     365
5526487     365
7449448     365
1776190     365
14501620    365
5967421     365
7540480     365
8067585     365
951480      365
11593592    365
11159585    365
1458081     365
7513906     365
9446242     365
8014383     365
2246959     365
13545648    365
5521378     365
4580825     365
8081935     365
9218288     365
8191739     365
6240901     365
14683493    365
951476      365
496022      365
13152370    365
531552      365
14668441    365
332225      365
4107959     365
2547129     365
6780300     365
5025015     365
13397201    365
9110588     365
6017574     365
10175033    365
6939759     365
13685577    365
14647971    365
6167184 

In [4]:
repeat = calendar.loc[calendar['listing_id'] == 12898806, ]
repeat.groupby(['listing_id', 'date']).count()

available  price
listing_id date                        
12898806   2016-09-06          2      2
           2016-09-07          2      2
           2016-09-08          2      2
           2016-09-09          2      2
           2016-09-10          2      2
           2016-09-11          2      2
           2016-09-12          2      2
           2016-09-13          2      2
           2016-09-14          2      0
           2016-09-15          2      0
           2016-09-16          2      0
           2016-09-17          2      0
           2016-09-18          2      0
           2016-09-19          2      0
           2016-09-20          2      0
           2016-09-21          2      0
           2016-09-22          2      0
           2016-09-23          2      0
           2016-09-24          2      2
           2016-09-25          2      2
           2016-09-26          2      2
           2016-09-27          2      2
           2016-09-28          2      2
           2016-09-29          2      2
           2016-09-30          2      2
           2016-10-01          2      2
           2016-10-02          2      2
           2016-10-03          2      2
           2016-10-04          2      2
           2016-10-05          2      2
           2016-10-06          2      2
           2016-10-07          2      2
           2016-10-08          2      2
           2016-10-09          2      2
           2016-10-10          2      2
           2016-10-11          2      2
           2016-10-12          2      2
           2016-10-13          2      2
           2016-10-14          2      2
           2016-10-15          2      2
           2016-10-16          2      2
           2016-10-17          2      2
           2016-10-18          2      2
           2016-10-19          2      2
           2016-10-20          2      2
           2016-10-21          2      2
           2016-10-22          2      2
           2016-10-23          2      2
           2016-10-24          2      2
           2016-10-25          2      2
           2016-10-26          2      2
           2016-10-27          2      2
           2016-10-28          2      2
           2016-10-29          2      2
           2016-10-30          2      2
           2016-10-31          2      2
           2016-11-01          2      2
           2016-11-02          2      2
           2016-11-03          2      2
           2016-11-04          2      2
           2016-11-05          2      2
           2016-11-06          2      2
           2016-11-07          2      2
           2016-11-08          2      2
           2016-11-09          2      2
           2016-11-10          2      2
           2016-11-11          2      2
           2016-11-12          2      2
           2016-11-13          2      2
           2016-11-14          2      2
           2016-11-15          2      2
           2016-11-16          2      2
           2016-11-17          2      2
           2016-11-18          2      2
           2016-11-19          2      2
           2016-11-20          2      2
           2016-11-21          2      2
           2016-11-22          2      2
           2016-11-23          2      2
           2016-11-24          2      2
           2016-11-25          2      2
           2016-11-26          2      2
           2016-11-27          2      2
           2016-11-28          2      2
           2016-11-29          2      2
           2016-11-30          2      2
           2016-12-01          2      2
           2016-12-02          2      2
           2016-12-03          2      2
           2016-12-04          2      2
           2016-12-05          2      0
           2016-12-06          2      0
           2016-12-07          2      0
           2016-12-08          2      0
           2016-12-09          2      0
           2016-12-10          2      0
           2016-12-11          2      0
           2016-12-12          2      0
           2016-12-13  

In [7]:
repeat

listing_id        date available    price
748250    12898806  2017-06-15         f      NaN
748251    12898806  2017-06-14         f      NaN
748252    12898806  2017-06-13         f      NaN
748253    12898806  2017-06-12         f      NaN
748254    12898806  2017-06-11         f      NaN
748255    12898806  2017-06-10         f      NaN
748256    12898806  2017-06-09         f      NaN
748257    12898806  2017-06-08         f      NaN
748258    12898806  2017-06-07         f      NaN
748259    12898806  2017-06-06         f      NaN
748260    12898806  2017-06-05         f      NaN
748261    12898806  2017-06-04         f      NaN
748262    12898806  2017-06-03         f      NaN
748263    12898806  2017-06-02         f      NaN
748264    12898806  2017-06-01         f      NaN
748265    12898806  2017-05-31         f      NaN
748266    12898806  2017-05-30         f      NaN
748267    12898806  2017-05-29         f      NaN
748268    12898806  2017-05-28         f      NaN
748269    12898806  2017-05-27         f      NaN
748270    12898806  2017-05-26         f      NaN
748271    12898806  2017-05-25         f      NaN
748272    12898806  2017-05-24         f      NaN
748273    12898806  2017-05-23         f      NaN
748274    12898806  2017-05-22         f      NaN
748275    12898806  2017-05-21         f      NaN
748276    12898806  2017-05-20         f      NaN
748277    12898806  2017-05-19         f      NaN
748278    12898806  2017-05-18         f      NaN
748279    12898806  2017-05-17         f      NaN
748280    12898806  2017-05-16         f      NaN
748281    12898806  2017-05-15         f      NaN
748282    12898806  2017-05-14         f      NaN
748283    12898806  2017-05-13         f      NaN
748284    12898806  2017-05-12         f      NaN
748285    12898806  2017-05-11         f      NaN
748286    12898806  2017-05-10         f      NaN
748287    12898806  2017-05-09         f      NaN
748288    12898806  2017-05-08         f      NaN
748289    12898806  2017-05-07         f      NaN
748290    12898806  2017-05-06         f      NaN
748291    12898806  2017-05-05         f      NaN
748292    12898806  2017-05-04         f      NaN
748293    12898806  2017-05-03         f      NaN
748294    12898806  2017-05-02         f      NaN
748295    12898806  2017-05-01         f      NaN
748296    12898806  2017-04-30         f      NaN
748297    12898806  2017-04-29         f      NaN
748298    12898806  2017-04-28         f      NaN
748299    12898806  2017-04-27         f      NaN
748300    12898806  2017-04-26         f      NaN
748301    12898806  2017-04-25         f      NaN
748302    12898806  2017-04-24         f      NaN
748303    12898806  2017-04-23         f      NaN
748304    12898806  2017-04-22         f      NaN
748305    12898806  2017-04-21         f      NaN
748306    12898806  2017-04-20         f      NaN
748307    12898806  2017-04-19         f      NaN
748308    12898806  2017-04-18         f      NaN
748309    12898806  2017-04-17         f      NaN
748310    12898806  2017-04-16         f      NaN
748311    12898806  2017-04-15         f      NaN
748312    12898806  2017-04-14         f      NaN
748313    12898806  2017-04-13         f      NaN
748314    12898806  2017-04-12         f      NaN
748315    12898806  2017-04-11         f      NaN
748316    12898806  2017-04-10         f      NaN
748317    12898806  2017-04-09         f      NaN
748318    12898806  2017-04-08         f      NaN
748319    12898806  2017-04-07         f      NaN
748320    12898806  2017-04-06         f      NaN
748321    12898806  2017-04-05         f      NaN
748322    12898806  2017-04-04         f      NaN
748323    12898806  2017-04-03         f      NaN
748324    12898806  2017-04-02         f      NaN
748325    12898806  2017-04-01         f      NaN
748326    12898806  2017-03-31         f      NaN
748327    12898806  2017-03-30         f      NaN
748328    12898806  2017-03-29         f      NaN
748329  

In [51]:
repeat.price.value_counts()

220.0    160
Name: price, dtype: int64

In [15]:
calendar.date.max()

'2017-09-05'

In [17]:
calendar.available.value_counts()

f    665853
t    643037
Name: available, dtype: int64

In [30]:
calendar.price.value_counts()

$150.00      20752
$65.00       17518
$75.00       16249
$100.00      15499
$200.00      14326
$70.00       13935
$60.00       12751
$80.00       12650
$175.00      11900
$250.00      11415
$125.00       9651
$85.00        9597
$99.00        9081
$199.00       8699
$300.00       8440
$90.00        8154
$225.00       7699
$275.00       7654
$120.00       7343
$299.00       7293
$349.00       7222
$55.00        7122
$110.00       6997
$249.00       6463
$135.00       6424
$160.00       6234
$130.00       6163
$500.00       6163
$95.00        5777
$350.00       5664
$79.00        5404
$155.00       5212
$45.00        4827
$50.00        4567
$165.00       4546
$145.00       4517
$89.00        4440
$115.00       4380
$69.00        4213
$195.00       4210
$399.00       4016
$329.00       3904
$180.00       3890
$49.00        3882
$129.00       3798
$189.00       3752
$220.00       3611
$140.00       3544
$295.00       3522
$185.00       3463
$59.00        3406
$40.00        3316
$179.00     

In [33]:
# clean currency data
calendar['price'] = calendar['price'].str.replace('$', '')
calendar['price'] = calendar['price'].str.replace(',', '')
calendar['price'] = calendar['price'].astype(float)
calendar.price.describe()

count    643037.000000
mean        198.438909
std         180.142913
min          11.000000
25%          85.000000
50%         150.000000
75%         255.000000
max        7163.000000
Name: price, dtype: float64

# Summary of calendar info
1. Calendar has time series info of different listing_id with their date, availability and price
2. each listing id has 365 observation(1 year), while id = 12898806 has replicated data; Its price remain unchanged for a whole year as 220, so just remove half of its data
3. date ranges from 2016-09-06 TO 2017-09-05
4. almost half of them are available, half of them are not available
5. price of listing ranges from 11 to 7163, mean is 198 and median is 180

# Explore what info listings contain

In [34]:
listings.shape

(3585, 95)

In [35]:
listings.head()

id                            listing_url       scrape_id  \
0  12147973  https://www.airbnb.com/rooms/12147973  20160906204935   
1   3075044   https://www.airbnb.com/rooms/3075044  20160906204935   
2      6976      https://www.airbnb.com/rooms/6976  20160906204935   
3   1436513   https://www.airbnb.com/rooms/1436513  20160906204935   
4   7651065   https://www.airbnb.com/rooms/7651065  20160906204935   

  last_scraped                                           name  \
0   2016-09-07                     Sunny Bungalow in the City   
1   2016-09-07              Charming room in pet friendly apt   
2   2016-09-07               Mexican Folk Art Haven in Boston   
3   2016-09-07  Spacious Sunny Bedroom Suite in Historic Home   
4   2016-09-07                            Come Home to Boston   

                                             summary  \
0  Cozy, sunny, family home.  Master bedroom high...   
1  Charming and quiet room in a second floor 1910...   
2  Come stay with a friendly, middle-aged guy in ...   
3  Come experience the comforts of home away from...   
4  My comfy, clean and relaxing home is one block...   

                                               space  \
0  The house has an open and cozy feel at the sam...   
1  Small but cozy and quite room with a full size...   
2  Come stay with a friendly, middle-aged guy in ...   
3  Most places you find in Boston are small howev...   
4  Clean, attractive, private room, one block fro...   

                                         description experiences_offered  \
0  Cozy, sunny, family home.  Master bedroom high...                none   
1  Charming and quiet room in a second floor 1910...                none   
2  Come stay with a friendly, middle-aged guy in ...                none   
3  Come experience the comforts of home away from...                none   
4  My comfy, clean and relaxing home is one block...                none   

                               neighborhood_overview  \
0  Roslindale is quiet, convenient and friendly. ...   
1  The room is in Roslindale, a diverse and prima...   
2  The LOCATION: Roslindale is a safe and diverse...   
3  Roslindale is a lovely little neighborhood loc...   
4  I love the proximity to downtown, the neighbor...   

                                               notes  \
0                                                NaN   
1  If you don't have a US cell phone, you can tex...   
2  I am in a scenic part of Boston with a couple ...   
3  Please be mindful of the property as it is old...   
4  I have one roommate who lives on the lower lev...   

                                             transit  \
0  The bus stop is 2 blocks away, and frequent. B...   
1  Plenty of safe street parking. Bus stops a few...   
2  PUBLIC TRANSPORTATION: From the house, quick p...   
3  There are buses that stop right in front of th...   
4  From Logan Airport  and South Station you have...   

                                              access  \
0  You will have access to 2 bedrooms, a living r...   
1  Apt has one more bedroom (which I use) and lar...   
2  I am living in the apartment during your stay,...   
3  The basement has a washer dryer and gym area. ...   
4  You will have access to the front and side por...   

                                         interaction  \
0                                                NaN   
1  If I am at home, I am likely working in my hom...   
2  ABOUT ME: I'm a laid-back, friendly, unmarried...   
3  We do live in the house therefore might be som...   
4  I love my city and really enjoy sharing it wit...   

                                         house_rules  \
0  Clean up and treat the home the way you'd like...   
1  Pet friendly but please confirm with me if the...   
2  I encourage you to use my kitchen, cooking and...   
3  - The bathroom and house are shared so please ...   
4  Please no smoking in the house, porch or on th...   

                                       thumb

In [37]:
listings.neighbourhood.value_counts()

Allston-Brighton      364
Jamaica Plain         314
South End             298
Back Bay              291
Fenway/Kenmore        249
South Boston          216
Dorchester            195
Beacon Hill           174
North End             125
East Boston           117
Roxbury               116
Mission Hill          103
Charlestown            79
Chinatown              78
West End               68
Roslindale             50
West Roxbury           35
Theater District       33
Downtown Crossing      26
Hyde Park              25
Mattapan               20
Somerville             13
Financial District     13
Downtown                8
Brookline               8
Leather District        8
Cambridge               7
Chestnut Hill           4
Government Center       3
Harvard Square          2
Name: neighbourhood, dtype: int64

# Summary of listings info
1. listings has info about id, listing url, last scraped, description, host, neighbourhood(zipcode), property info

# Explore what info reviews contain

In [3]:
reviews.shape

(68275, 6)

In [4]:
reviews.head()

listing_id       id        date  reviewer_id reviewer_name  \
0     1178162  4724140  2013-05-21      4298113       Olivier   
1     1178162  4869189  2013-05-29      6452964     Charlotte   
2     1178162  5003196  2013-06-06      6449554     Sebastian   
3     1178162  5150351  2013-06-15      2215611        Marine   
4     1178162  5171140  2013-06-16      6848427        Andrew   

                                            comments  
0  My stay at islam's place was really cool! Good...  
1  Great location for both airport and city - gre...  
2  We really enjoyed our stay at Islams house. Fr...  
3  The room was nice and clean and so were the co...  
4  Great location. Just 5 mins walk from the Airp...

In [5]:
reviews.date.max()

'2016-09-06'

# Summary of reviews info
1. reviews has info about listing, reviewer, and their comments
2. reviews includes 68k comments from to 2009-03-21 to 2016-09-06.